projects/aws/bottlerocket-bootstrap/pkg/providers/snow/system/network.go

```diff
@@ -0,0 +1,107 @@
+package system
+
+import (
+	_ "embed"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"path/filepath"
+	"syscall"
+
+	"github.com/pkg/errors"
+	"github.com/vishvananda/netlink"
+
+	"github.com/eks-anywhere-build-tooling/aws/bottlerocket-bootstrap/pkg/files"
+)
+
+//go:embed config/net.toml
+var netConfig string
+
+const (
+	rootfs            = "/.bottlerocket/rootfs"
+	metadataServiceIP = "169.254.169.254"
+)
+
+var netConfigPath = filepath.Join(rootfs, "/var/lib/netdog/net.toml")
+
+func configureDNI() error {
+	if files.PathExists(netConfigPath) {
+		return nil
+	}
+
+	instanceIP, err := instanceIP()
+	if err != nil {
+		return errors.Wrap(err, "error getting local instance IP")
+	}
+
+	dnis, err := dniList()
+	if err != nil {
+		return errors.Wrap(err, "error getting DNI list")
+	}
+
+	defaultGateway, err := defaultGateway()
+	if err != nil {
+		return errors.Wrap(err, "error getting default gateway")
+	}
+
+	val := map[string]interface{}{
+		"dnis":           dnis,
+		"instanceIP":     instanceIP,
+		"defaultGateway": defaultGateway,
+	}
+
+	if err := files.WriteTemplate(netConfigPath, netConfig, val, 0o640); err != nil {
+		return errors.Wrapf(err, "error writing network configuration to %s", netConfigPath)
+	}
+
+	return nil
+}
+
+func dniList() ([]string, error) {
+	devices, err := netlink.DevLinkGetDeviceList()
+	if err != nil {
+		return nil, errors.Wrap(err, "error getting the device list of links")
+	}
+	dnis := []string{}
+	for _, device := range devices {
+		name := device.DeviceName
+		if name != "lo" && name != "eth0" {
+			dnis = append(dnis, name)
+		}
+	}
+	return dnis, nil
+}
+
+func defaultGateway() (string, error) {
+	routes, err := netlink.RouteList(nil, syscall.AF_INET)
+	if err != nil {
+		return "", errors.Wrap(err, "error getting the route list")
+	}
+
+	for _, route := range routes {
+		if route.Dst == nil {
```

Comment:
what is this route.Dst?

Reply:
it's IPNet and `== nil` represents the default gateway